src/devices/Seesaw/SeesawBase.cs

```diff
@@ -0,0 +1,176 @@
+        protected void WriteByte(Seesaw_Module moduleAddress, Seesaw_Function functionAddress, byte value)
+        {
+            Write(moduleAddress, functionAddress, new byte[] { value });
+        }
+
+        /// <summary>
+        /// Write a series of bytes to the I2cDevice connected to the Seesaw board. 
+        /// </summary>
+        /// <param name="moduleAddress">An Seesaw_Module enum that represents the module that we are writing to.</param>
+        /// <param name="functionAddress">An Seesaw_Function enum that represents the Seesaw function to be called.</param>
+        /// <param name="data">The byte array containg data to be send as a parameter or parameters to the Seesaw device.</param>
+        protected void Write(Seesaw_Module moduleAddress, Seesaw_Function functionAddress, byte[] data)
+        {
+            const int StackThreshold = 512;
+
+            Span<byte> buffer = data.Length < StackThreshold ? stackalloc byte[data.Length + 2] : new byte[data.Length + 2];
+
+            for (int i = 0; i < data.Length; buffer[i + 2] = data[i], i++) ;
+
+            buffer[0] = (byte)moduleAddress;
+            buffer[1] = (byte)functionAddress;
+            I2cDevice.Write(buffer);
+        }
+
+        /// <summary>
+        /// Read a byte from the I2cDevice connected to the Seesaw board. 
+        /// </summary>
+        /// <param name="moduleAddress">A Seesaw_Module enum that represents the module that we are reading from.</param>
+        /// <param name="functionAddress">A Seesaw_Function enum that represents the Seesaw function to be called.</param>
+        /// <param name="delay">The delay between sending the function all and readinng the data in microseconds.</param>
+        /// <returns>Returns the byte value from the Seesaw device.</returns>
+        protected byte ReadByte(Seesaw_Module moduleAddress, Seesaw_Function functionAddress, short delay = 0)
+        {
+            return (Read(moduleAddress, functionAddress, 1, delay)[0]);
+        }
+
+        /// <summary>
+        /// Read a byte array from the I2cDevice connected to the Seesaw board. 
+        /// </summary>
+        /// <param name="moduleAddress">A Seesaw_Module enum that represents the module that we are reading from.</param>
+        /// <param name="functionAddress">A Seesaw_Function enum that represents the Seesaw function to be called.</param>
+        /// <param name="length">The number of bytes that are expected to be returned from the Seesaw device.</param>
+        /// <param name="delay">The delay between sending the function all and readinng the data in microseconds.</param>
+        /// <returns>Returns the byte array representaing values from the Seesaw device.</returns>
+        protected byte[] Read(Seesaw_Module moduleAddress, Seesaw_Function functionAddress, int length, short delay = 0)
+        {
+
+            byte[] retval = new byte[length];
```

Comment:
better solution would be taking `Span<byte>` as an input to the function (return `void` and `length` is implied by `Span`)

Reply:
@krwq 
 But that would add complexity to the callers of the function. So for example, the following

`return Read(moduleAddress, functionAddress, 1, delay)[0];`

would need to be replaced by...

```
 Span<byte> buffer = stackalloc byte[2];
 Read(moduleAddress, functionAddress, buffer, delay);
 return buffer[0];

```